File: Java/src/main/java/QuestionBank/Solution2301_2400/Solution2367.java
package QuestionBank.Solution2301_2400;

import java.util.HashSet;

/**
 * @author windows
 */
public class Solution2367 {
    public int arithmeticTriplets(int[] nums, int diff) {
        HashSet<Integer> set = new HashSet<>();
        for (int i = 0; i < nums.length; i++) {
            set.add(nums[i]);
        }
        int count = 0;
        for (int i = 0; i < nums.length; i++) {
            if (set.contains(nums[i] + diff) && set.contains(nums[i] + diff * 2)) {
                count++;
            }
        }
        return count;
    }
    
    
    public static void main(String[] args) {
        Solution2367 solution2367 = new Solution2367();
    }
}

File: C++/src/Solution2301_2400/Solution2367.cpp
//
// Created by windows on 2023/3/2.
//

#include "Solution2367.h"
#include <functional>
#include <iostream>
#include <vector>
#include <unordered_set>
#include <unordered_map>
#include <string>
using namespace std;
#include "vector"
#include <algorithm>
using namespace std;


int arithmeticTriplets(vector<int>& nums, int diff) {
  int n = nums.size();
  int count = 0;
  for (int i = 2; i < n; ++i) {
    for (int j = 1; j < i; ++j) {
      for (int k = 0; k < j; ++k) {
        if (nums[i] - nums[j] == diff && nums[j] - nums[k] == diff){
          count++;
        }
      }
    }
  }
  return count;
}
